cmd/grpcurl/grpcurl.go

```diff
@@ -486,6 +483,13 @@ func main() {
 		if *maxMsgSz > 0 {
 			opts = append(opts, grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(*maxMsgSz)))
 		}
+		if isUnixSocket != nil && isUnixSocket() && !strings.HasPrefix(target, "unix://") {
+			// prepend unix:// to the address if it's not already there
+			// this is to maintain backwards compatibility because the custom dialer is replaced by
+			// the default dialer in grpc-go.
+			// https://github.com/fullstorydev/grpcurl/pull/480
+			target = "unix://" + target
+		}
```

Comment:
You could instead set a `network` local variable (that defaults to empty string) to `"unix"` and then pass to `BlockingDial` below. That way, you aren't duplicating the logic that also lives in `BlockingDial` for this scenario.